src/kernel_2.jl

```diff
@@ -1,27 +1,40 @@
 function key_sampling(
-  g::SimpleWeightedGraph{T,R}
-  ) where T<:Integer where R<:Real
+  g::Graph{T}
+  ) where T<:Integer
 
   n_v = nv(g)
-  rand_keys = sample(1:n_v, 64, replace = false)
+  rand_keys = randperm(n_v)
   keys = Vector{T}()
   sizehint!(keys, 64)
 
-  for i in 1:64
-    if degree(g, rand_keys[i]) > 0
-      push!(keys, rand_keys[i])
+  for i in 1:n_v
+    if length(keys) == 64
```

Comment:
NBFS is defined in `key_sampling()` as ` NBFS = n_v > 64 ? 64 : n_v`. But also it is possible that 64 keys with `degree(key) > 0` might not exist on graph with `n_v > 64`. So I will have to terminate the loop after a few fixed number of calls to `sample` , else it will go in infinite loop.
So NBFS is just the number of keys in the output of `key_sampling()`